cpp/scripts/gitutils.py

```diff
@@ -0,0 +1,134 @@
+# Copyright (c) 2019, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import subprocess
+import os
+
+
+def isFileEmpty(f):
+    return os.stat(f).st_size == 0
+
+
+def __git(*opts):
+    """Runs a git command and returns its output"""
+    cmd = "git " + " ".join(list(opts))
+    return subprocess.check_output(cmd, shell=True)
+
+
+def __gitdiff(*opts):
+    """Runs a git diff command with no pager set"""
+    return __git("--no-pager", "diff", *opts)
+
+
+def branch():
+    """Returns the name of the current branch"""
+    name = __git("rev-parse", "--abbrev-ref", "HEAD")
+    name = name.rstrip()
+    return name
+
+
+def uncommittedFiles():
+    """
+    Returns a list of all changed files that are not yet committed. This
+    means both untracked/unstaged as well as uncommitted files too.
```

Comment:
So, an untracked file possibly not used at all in the build could fail the build then? Like if I had a stray file hanging out in a subdir of my repo? (I have to admit I do this occasionally with a file like `scratch.py` or `junk.cpp` that I'm using for notes and snippets, though I'm open to trying to change old habits.)

What if we just applied this to files that are uncommitted and staged? Then it's only going to hit files developers really care about.

Reply:
If we didn't also look at unstaged files, we might miss out on files that are being newly added to the repo, only to be caught during the CI process inside PR. Hence, to make sure devs will not have any such surprises, I didn't want to take any chances. Thus, it covers all c++/cuda files that have been just added, not been committed as well as staged to be committed. What say?

Reply:
After our discussion last night between you, myself and Dante, it makes sense to not touch untracked files. Have updated the same.